lib/Runtime/Library/JavascriptString.cpp

```diff
@@ -2134,37 +2134,51 @@ namespace Js
     Var JavascriptString::ToCaseCore(JavascriptString* pThis, ToCase toCase)
     {
         charcount_t count = pThis->GetLength();
+        charcount_t countToCase = count;
 
+        const wchar_t* inStr = pThis->GetString();
+        const wchar_t* inStrLim = inStr + count;
+        const wchar_t* i = inStr;
+
+        // Try to find out the chars that do not need casing
+        if (toCase == ToUpper)
+        {
+            while (i < inStrLim) {
+                if (*i >= 224 || (*i <= 118 && *i >= 97)) break;
+                i++;
+            }
+        } else {
+            Assert(toCase == ToLower);
+            while (i < inStrLim) {
+                if (*i >= 257 || (*i <= 204 && *i >= 192) || (*i <= 90 && *i >= 65)) break;
+                i++;
+            }
+        }
```

Comment:
I have a few comments here.
- the style doesn't match our C++ style convention
  - braces should be on newlines
  - if consequent etc must be braced blocks, e.g. `if (...) { break; }` in this case
- instead of integer literals for the ranges can you use character literals
- and can you reverse the conditions so that lower numbers are left most and the `*i` reads in between the bounds
  - e.g. `if (('a' <= *i && *i <= 'z') || *i >= 'à') { break; }`
- 118 is `'v'`, you meant 122 `'z'` right?
- I'm presuming these ranges match the behavior of `CharUpperBufW` and `CharLowerBufW`, which should match the unicode spec, but I do not know quickly offhand where to look that up.
  - Can you provide a comment that states this is a fast path for strings that lie within the ascii range and happen to have only characters that already match the desired casing, and provide a link to a spec reference that clearly shows these are the right ranges